apps/grep.c

```
@@ -6,45 +6,52 @@
-        if (n <= 0) break;
-        total += n;
+    } else {
+        capacity = 512;
+        data = malloc(capacity);
+        if (!data) { print("grep: out of memory\n"); return 1; }
+        char buf[512];
+        int n;
+        while ((n = sys_read_fd(0, buf, sizeof(buf))) > 0) {
+            while (total + n > capacity) {
+                capacity *= 2;
+                data = realloc(data, capacity);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)